core/src/main/java/com/datastrato/gravitino/catalog/CatalogOperationDispatcher.java

```diff
@@ -446,12 +447,19 @@ public Table createTable(
     NameIdentifier catalogIdent = getCatalogIdentifier(ident);
     doWithCatalog(
         catalogIdent,
-        c ->
-            c.doWithPropertiesMeta(
-                p -> {
-                  validatePropertyForCreate(p.tablePropertiesMetadata(), properties);
-                  return null;
-                }),
+        c -> {
+          c.doWithPropertiesMeta(
+              p -> {
+                validatePropertyForCreate(p.tablePropertiesMetadata(), properties);
+                return null;
+              });
+          c.doWithCapabilities(
```

Comment:
I was thinking if we can have a new class to apply all the capabilities, not split into everywhere in this class, that will be easy to manage and add new capability. Like Spark's `Strategy`.

Reply:
I will try to strip it off to slim down CatalogOperationDispatcher.java.